fireapp/mock-data/shift_request.json

```diff
@@ -0,0 +1,47 @@
+{
+  "results": [
```

Comment:
Your models can't currently parse this, could you please create a generic type parameter model to hold results? It would be called Results, accept a single type parameter, and have a single field called "Results" of that type parameter.

Reply:
How do I do that?

Reply:
```dart
class Result<T> {
  
}
```

with a field of `T result`. Do the whole freezed thing. If that doesn't compile tell me

Reply:
@freezed
class Results<T> with _$Results<T> {
  const factory Results({
    @JsonKey(name: "results") required List<T> results,
  }) = _Results<T>;

  factory Results.fromJson(Map<String, Object?> json, T Function(Object? json) fromJsonT) => _$ResultsFromJson(json, fromJsonT);
}

Reply:
but it wont compile